Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m

```diff
@@ -411,16 +384,17 @@ - (UIFont *)dateHeaderTimeFont
 
 #pragma mark - Measurement
 
-- (CGSize)cellSizeForViewWidth:(int)viewWidth contentWidth:(int)contentWidth
+- (CGSize)cellSizeWithTransaction:(YapDatabaseReadTransaction *)transaction
 {
+    OWSAssert(self.layoutInfo);
+    OWSAssert(self.layoutInfo.viewWidth > 0);
     OWSAssert(self.viewItem);
     OWSAssert([self.viewItem.interaction isKindOfClass:[TSMessage class]]);
     OWSAssert(self.messageBubbleView);
 
     self.messageBubbleView.viewItem = self.viewItem;
-    self.messageBubbleView.contentWidth = self.contentWidth;
     self.messageBubbleView.cellMediaCache = self.delegate.cellMediaCache;
-    CGSize messageBubbleSize = [self.messageBubbleView sizeForContentWidth:contentWidth];
+    CGSize messageBubbleSize = [self.messageBubbleView measureSize];
```

Comment:
I was surprised that `measureSize` didn't ever need a transaction to compute it's sizing, but it looks like everything it uses is already cached on the viewModel. 👍

Reply:
Yup, that's where I'd like to get for all of this.